### ISSUE TITLE
Make default target result size configurable in ExecutingStatementResource

### ISSUE THREAD

**Issue by tooptoop4** [trinodb/trino#2210]
https://github.com/prestodb/presto/issues/11908 mentions targetresultsize, is anyone open to a pr to make it configurable in server side properties? ie so folks can try 4mb instead of 1mb. returning 1gb data into reporting tool with odbc/jdbc is really slow. dumping to file on presto server is fast but not an option because the reporting tool cant access files on presto server

**Comment by electrum** [comment]
Let's add a `query.target-result-size.default` config option

**Comment by tooptoop4** [comment]
what classes should be modified for this? especially about where the @Config gets read in. i thought  
  SqlEnvironmentConfig,     QuerySessionSupplier ,   ExecutingStatementResource would be involved but Session does not accept argument for targetResultSize

**Comment by tooptoop4** [comment]
https://github.com/prestosql/presto/pull/2347

**Comment by tooptoop4** [comment]
would be good to log the final targetResultSize value in v1/query json

**Comment by hashhar** [comment]
@electrum Would the approach in https://github.com/prestodb/presto/pull/14641 of adding a session property/config and *additionally having Presto itself add the targetResultSize parameter to `nextUri`* be welcome change?

This will make this property transparent to all clients and will be helpful in deployments where we don't control all clients or the clients don't make it easy to pass custom properties or the change required forking the client.

What does everybody think?

To summarize:

1. A config `query.target-result-size.default` with a matching session property.
2. Presto uses the config/session property to adjust the nextUri returned via https://github.com/prestosql/presto/blob/62cf1510d7cec2cd762711b6b3358d7e3259ac65/presto-main/src/main/java/io/prestosql/server/protocol/Query.java#L582.
3. Both JDBC and HTTP clients enjoy :confetti_ball:

**Comment by sajjoseph** [comment]
@hashhar - I like the idea of using the session variable to control the behavior.
I had implemented the changes for CLI and JDBC driver. But, now that I think about it, setting the session variable might be easier with existing clients. 

I will update my PR #4347 to include this change.

**Comment by itguruhaseeb** [comment]
The linked PRs (#2347, #4347) appear to have gone quiet a while back. If nobody is actively on this, I'd like to pick it up following the approach @electrum and @hashhar outlined: a query.target-result-size.default config with a matching session property, applied when building nextUri in the protocol layer so it works transparently for all clients. I have PRs in review for #4697 (PR #30737) and #5137 (PR #30750), so I'm familiar with the contribution flow. If the earlier authors are still planning to finish theirs, happy to step aside.

**Comment by sajjoseph** [comment]
With the spooling support, there are other ways to address the need. Are there any features that spooling protocol can't handle that this feature will address?
